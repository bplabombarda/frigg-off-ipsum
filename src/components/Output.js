import React from 'react';
import { array } from 'prop-types'

import '../styles/Output.scss';

const Output = ({ paragraphs }) => {
	return (
		<section className='output-container'>
			<div className={ paragraphs.length ? 'output' : 'output hidden' } >
				{
					paragraphs.map((paragraph, index) => (
						<p key={ index }>{ paragraph }</p>
					))
				}
			</div>
		</section>
	);
};

Output.propTypes = {
  paragraphs: array,
};

export default Output;
