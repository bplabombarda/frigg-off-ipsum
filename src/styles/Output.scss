#output-text {
	font-size: 0.8em;
	max-height: 360px;
	margin: 0 auto;
	padding: 1em;
	background: white;	
	overflow: scroll;
}

::-webkit-scrollbar {
	-webkit-appearance: none;
	width: 7px;
}

::-webkit-scrollbar-thumb {
	border-radius: 4px;
	background-color: rgba(0,0,0,.5);
	-webkit-box-shadow: 0 0 1px rgba(255,255,255,.5);
}
