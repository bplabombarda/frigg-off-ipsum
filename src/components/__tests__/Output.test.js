import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import Output from '../Output';

const mockProps = {
  paragraphs: [],
};

describe('rendering', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Output { ...mockProps } />
    );
  
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});