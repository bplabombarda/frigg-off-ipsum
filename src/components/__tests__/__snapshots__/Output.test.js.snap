// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering matches snapshot 1`] = `
<section
  className="output-container"
>
  <div
    className="output hidden"
  />
</section>
`;
